src/types/monte_carlo.rs

```diff
@@ -552,48 +552,48 @@ impl<'a> VegasParams<'a> {
                stream: Option<&'a mut ::IOStream>)
                -> Result<VegasParams, String> {
         if !verbosity.is_off() && stream.is_none() {
-            return Err("rust-GSL: ".to_string());
+            return Err("rust-GSL: need to provide an input stream for Vegas Monte Carlo \
+                        integration if verbosity is not 'Off'"
+                .to_string());
         } else if verbosity.is_off() && stream.is_some() {
-            return Err("rust-GSL: ".to_string());
+            return Err("rust-GSL: need to provide the verbosity flag for Vegas Monta Carlo \
```

Comment:
in the first condition verbosity flag is not Off (-1) but an input stream (file really) is not provided;
   this would result in UB in C as would be pointing to a null ptr
in the second one verbosity flag is Off(-1) but some input stream was provided;
  this could be ignored but is probably undesired behavior from the function caller

in practice everything could be resolved in a single condition, but the error message would be more ambiguous.

ideally we would have a proper error chain too, is suboptimal to return strings imo, but considering that there are not many cases (so far) were we would be using them (for now at least) this should do for now.